aes-siv/src/lib.rs

```diff
@@ -105,20 +105,21 @@ use digest::{FixedOutputReset, Mac};
 use pmac::Pmac;
 
 /// AES-SIV nonces
-pub type Nonce = GenericArray<u8, U16>;
+pub type Nonce<NonceSize = U16> = GenericArray<u8, NonceSize>;
 
 /// AES-SIV tags (i.e. the Synthetic Initialization Vector value)
 pub type Tag = GenericArray<u8, U16>;
 
 /// The `SivAead` type wraps the more powerful `Siv` interface in a more
 /// commonly used Authenticated Encryption with Associated Data (AEAD) API,
 /// which accepts a key, nonce, and associated data when encrypting/decrypting.
-pub struct SivAead<C, M>
+pub struct SivAead<C, M, NonceSize = U16>
 where
     Self: KeySizeUser,
     C: BlockCipher<BlockSize = U16> + BlockEncryptMut + KeyInit + KeySizeUser,
     M: Mac<OutputSize = U16> + FixedOutputReset + KeyInit,
     <C as KeySizeUser>::KeySize: Add,
+    NonceSize: ArrayLength<u8>,
```

Comment:
From [RFC5297 § 6.1](https://www.rfc-editor.org/rfc/rfc5297.html#section-6.1):

```
 N_MIN  is 1 octet.
```

So this should be reflected in the bounds, and probably documented somewhere:

```suggestion
    NonceSize: ArrayLength<u8> + IsGreaterOrEqual<U1>
```

Reply:
Also:

> And the SivAead<C, M, NonceSize = U16> is not a specific AEAD algorithm mentioned in the RFC5297

The intent of `SivAead` is to implement the RFC5297 API, although the documentation to that effect may not be great.